R/marg.R

```diff
@@ -96,20 +106,46 @@
 #' marg(mod, var_interest = 'treatment', type = 'levels',
 #'           vcov_mat = v, dof = d)
 #'
+#' # Using weights
+#'
+#' data(margex)
+#' mm <- glm(y ~ as.factor(treatment) + age, data = margex, family = 'gaussian',
+#'           weights = distance)
+#' z1 <- marg(mod = mm, var_interest = 'treatment', type = 'levels')[[1]]
+#' z2 <- marg(mod = mm, var_interest = 'treatment', type = 'effects')[[1]]
+#'
 marg <- function(mod, var_interest,
-                      type = 'levels',
-                      vcov_mat = NULL,
-                      dof = NULL,
-                      at = NULL, base_rn = 1,
-                      at_var_interest = NULL,
-                      data = mod$data,
+                 type = 'levels',
+                 vcov_mat = NULL,
+                 dof = NULL,
+                 at = NULL, base_rn = 1,
+                 at_var_interest = NULL,
+                 data = mod$data,
+                 weights = mod$prior.weights,
                  cofint = 0.95){
 
   stopifnot('glm' %in% class(mod))
 
+  # Remove weights if no weights
+  if(all(weights == 1)) weights <- NULL
+
+  # Check if no weights when model was built was weights
+  if(is.null(weights) & !all(mod$prior.weights == 1))
+    warning(paste('The model was built with weights, but you have not',
+                  'provided weights. Your calculated margins may be odd.',
+                  'See Details.'))
+
+  # Weights should be same length as data
+  if(!is.null(weights) & length(weights) != nrow(data))
+    stop('`weights` and `data` must be the same length.')
```

Comment:
Shouldn't this check happen after data and weights are subset to `complete_cases`?

Otherwise somebody could pass `data` with no missingness and `weights` with missingness, and the `complete_cases` stuff below wouldn't handle missingness in the weights.